classes/webservice/WebserviceOutputBuilder.php

```diff
@@ -378,131 +384,93 @@ public function getContent($objects, $schema_to_display = null, $fields_to_displ
     {
         $this->fieldsToDisplay = $fields_to_display;
         $this->depth = $depth;
-        $output = '';
 
         if ($schema_to_display != null) {
             $this->schemaToDisplay = $schema_to_display;
-            $this->objectRender->setSchemaToDisplay($this->schemaToDisplay);
-
-            // If a shema is asked the view must be an details type
-            $type_of_view = self::VIEW_DETAILS;
-        }
-
-        $class = get_class($objects['empty']);
-        if (!isset(WebserviceOutputBuilder::$_cache_ws_parameters[$class])) {
-            WebserviceOutputBuilder::$_cache_ws_parameters[$class] = $objects['empty']->getWebserviceParameters();
+            $type_of_view = self::VIEW_DETAILS; // If a shema is asked the view must be an details type
         }
-        $ws_params = WebserviceOutputBuilder::$_cache_ws_parameters[$class];
+        $ws_params = $this->loadWsParams($objects['empty']);
 
         foreach ($this->wsParamOverrides as $p) {
             $object = $p['object'];
             $ws_params = $object->{$p['method']}($ws_params);
         }
 
-        // If a list is asked, need to wrap with a plural node
-        if ($type_of_view === self::VIEW_LIST) {
-            $output .= $this->setIndent($depth) . $this->objectRender->renderNodeHeader($ws_params['objectsNodeName'], $ws_params);
-        }
+        $this->output = $type_of_view == self::VIEW_DETAILS ? ApiNode::parent($ws_params['objectsNodeName']) : ApiNode::list($ws_params['objectsNodeName']);
 
         if (null === $this->schemaToDisplay) {
             foreach ($objects as $key => $object) {
                 if ($key !== 'empty') {
                     if ($this->fieldsToDisplay === 'minimum') {
-                        $output .= $this->renderEntityMinimum($object, $depth);
+                        $this->renderEntityMinimum($object);
                     } else {
-                        $output .= $this->renderEntity($object, $depth);
+                        $this->renderEntity($this->output, $object);
                     }
                 } elseif ($key == 'empty' && $this->objectRender->getContentType() == 'application/json') {
                     $output .= $this->renderEntity($object, $depth);
```

Comment:
```suggestion
                    $this->output .= $this->renderEntity($object, $depth);
```

Reply:
Fixed in 601c8f1 . Actually rendering functions are being called using
`$this->renderEntity(&$parentNode, ...): void`
So there is no appending to output - instead output is passed as the parameter. During this I also typed the renderEntity function to make it more clear and fixed one bug